components/authz-service/server/v2/project_update_manager.go

```diff
@@ -17,351 +17,424 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
-// TODO
-// Add a partial update state - when an update is canceled or failed
+// Failure cases
+// * No status event for 5 minutes
+// * Domain service sends failure message
+
+// Recovery
+// * Domain service was down during start
+// * Authz service restarts
+
 const (
-	RunningState    = "running"
-	NotRunningState = "not_running"
+	minutesWithoutCheckingInFailure = 5
+	sleepTimeBetweenStatusChecksSec = 5
 )
 
-type domainService struct {
-	name                   string
-	percentageComplete     float64
-	estimatedTimeCompelete time.Time
-	lastUpdate             time.Time
-	complete               bool
-	failed                 bool
+type update struct {
+	fun  func(config.ProjectUpdateStage) (config.ProjectUpdateStage, error)
+	errc chan error
 }
 
 // ProjectUpdateManager - project update manager
 type ProjectUpdateManager struct {
-	state              string
-	projectUpdateID    string
+	stage              config.ProjectUpdateStage
 	eventServiceClient automate_event.EventServiceClient
-	domainServices     []*domainService
+	configManager      *config.Manager
+	updateQueue        chan<- update
```

Comment:
This is a queue of updates to the ProjectUpdateStage. These change will only be processed in order by one thread.